### ISSUE TITLE
Option to avoid creating or updating a channel

### ISSUE THREAD

**Issue by gnprice** [MaikuB/flutter_local_notifications#2135]
This is a feature request. I'd be happy to send a PR for it if that would be welcome.

On Android, I prefer to create notification channels up front and manage them explicitly, rather than have them implicitly created or updated when I create a notification. If the channel doesn't yet exist, that's a bug in the app and I'd rather get an exception than implicitly create the channel — I'd rather drop that notification than create the channel with potentially different settings from usual, which could affect future notifications for a long time.

This preference also corresponds to the recommendations from Android upstream, which are to create the channel [as soon as your app starts](https://developer.android.com/develop/ui/views/notifications/build-notification#Priority), rather than at the time a notification arises.

I see there's already the `AndroidNotificationDetails.channelAction` property to control this area of behavior, with the enum `AndroidNotificationChannelAction`. The existing values are:
* `createIfNotExists` — `NotificationManagerCompat.createNotificationChannel` will be called if the channel does not already exist.
* `update` — `NotificationManagerCompat.createNotificationChannel` will be called if the channel already exists.

So I think what I want here is just to have a third value in that enum, with the meaning:
* `NotificationManagerCompat.createNotificationChannel` will not be called, regardless of whether the channel already exists.

and perhaps with a name like `neverCreate`.

Does that sound like a PR you'd be glad to take?

**Comment by MaikuB** [comment]
There's an existing issue related to this at https://github.com/MaikuB/flutter_local_notifications/issues/1528. Note though that the thought was to complete drop the channel-specific properties from `AndroidNotificationDetails` though there may be some double up as some properties shifted to being associated with a channel from Android 8 onwards (e.g. see https://developer.android.com/reference/androidx/core/app/NotificationCompat.Builder#setSound(android.net.Uri)). Would appreciate help with a PR on this if you're up for it

**Comment by gnprice** [comment]
Ah, I see that issue now. Yeah, so if I understand the idea correctly, that would effectively mean the `neverCreate` behavior would be the only behavior, and the existing `createIfNotExists` and `update` behaviors would go away.

I'd certainly be happy with that result. I guess I proposed a more cautious change mostly because I wasn't sure how willing you might be to break compatibility for existing apps. Maybe a little bit also because the more radical change will be somewhat more work. :-)

At the moment I'm busy finishing work for an app release that's a few weeks out. But after that I may be able to take a look.

**Comment by MaikuB** [comment]
That's right though there would still need to be some level of compatibility would still need to exist on the Java code side as apps may have existing scheduled notifications. The breaking changes would be more on the Dart API side

**Comment by Levi-Lesches** [comment]
Seems like this can be closed in favor of #1528
